packages/react/src/drafts/InlineAutocomplete/InlineAutocomplete.tsx

```diff
@@ -118,7 +118,7 @@ const InlineAutocomplete = ({
   const inputProps = useFormControlForwardedProps(externalInputProps)
 
   const inputRef = useRef<HTMLInputElement & HTMLTextAreaElement>(null)
-  useRefObjectAsForwardedRef(children.ref ?? noop, inputRef)
+  useRefObjectAsForwardedRef(children.ref && typeof children.ref !== 'string' ? children.ref : noop, inputRef)
```

Comment:
The string comparison seems like a bonus here, did this show up as validation error with the new version?

Reply:
@siddharthkp yeah! Just showed up as a TS error